hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -690,13 +705,8 @@ public ReplicationQueueInfo getReplicationQueueInfo() {
     return replicationQueueInfo;
   }
 
-  public boolean isWorkerRunning(){
-    for(ReplicationSourceShipper worker : this.workerThreads.values()){
-      if(worker.isActive()){
-        return worker.isActive();
-      }
-    }
-    return false;
+  public boolean isWorkerRunning() {
+    return !this.workerThreads.isEmpty();
```

Comment:
isn't this a change of semantics? (there could be window where it isn't active but still in the map..). Doesn't probably matter that much but just wondering..